pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -213,11 +213,11 @@ func (r *ReconcileUserSignup) provisionMasterUserRecord(userSignup *toolchainv1a
 
 	mur := &toolchainv1alpha1.MasterUserRecord{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      userSignup.Name,
+			Name:      userSignup.Spec.CompliantUsername,
```

Comment:
There are two things to be fixed:
1. when checking if the MUR already exists [here](https://github.com/codeready-toolchain/host-operator/pull/91/files#diff-5b4b18a0fc2e0d708e4e6334cdb6bc69R130) you should use `userSignup.Spec.CompliantUsername` in the namespaced name
2. because we are now using `compliantUsername` created on our own that is not unique, it can easily happen that such a MUR already exists so we would end-up here https://github.com/codeready-toolchain/host-operator/pull/91/files#diff-5b4b18a0fc2e0d708e4e6334cdb6bc69R133-R136. In that case, we would need to modify the name a bit (eg. ad suffix `-2`) to get the unique one

Reply:
@sbryzak I see that you fixed the first point but the second point is still missing. Could you please shed some light on why you don't do such a check or modification of MUR name? 
If the task is postponed to be done in a separated PR, do we have a Jira issue created for that?